src/app/space/settings/areas/areas.component.ts

```diff
@@ -50,10 +51,17 @@ export class AreasComponent implements OnInit, OnDestroy {
   subscriptions: Subscription[] = [];
   resultsCount: number = 0;
   treeListConfig: TreeListConfig;
-
-  constructor(private contexts: ContextService,
-              private areaService: AreaService) {
-    this.contexts.current.subscribe(val => this.context = val);
+  userOwnsSpace: boolean;
+
+  constructor(
+    private contexts: ContextService,
+    private areaService: AreaService,
+    private userService: UserService
+  ) {
+    this.contexts.current.subscribe((context: Context) => {
```

Comment:
Missing unsubscribe.

Reply:
Done. Also, changed the way subscriptions are handled now. There is one master subscription called `subscriptions` which is of type `Subscription` now which adds every `subscribe` call to it and then just one `unsubscribe()` call in `ngOnDestroy`. This is how `Ben Lesh` suggested we should handle multiple subscriptions in a JSFoo talk that i attended some time back.